pkg/handlers/lifecycle/konnectoragent/handler.go

```diff
@@ -651,3 +670,112 @@ func (n *DefaultKonnectorAgent) checkCleanupStatus(
+	clusterConfigVar, err := variables.Get[apivariables.ClusterConfigSpec](
+		varMap,
+		v1alpha1.ClusterConfigVariableName,
+	)
+	if err != nil {
+		return false, fmt.Errorf("failed to read clusterConfig variable: %w", err)
+	}
+
+	if clusterConfigVar.Nutanix == nil || clusterConfigVar.Nutanix.PrismCentralEndpoint.URL == "" {
+		return false, fmt.Errorf("prism central endpoint not configured")
+	}
+
+	prismCentralEndpointSpec := clusterConfigVar.Nutanix.PrismCentralEndpoint
+	host, port, err := prismCentralEndpointSpec.ParseURL()
+	if err != nil {
+		return false, fmt.Errorf("failed to parse prism central endpoint URL: %w", err)
+	}
+
+	// Get konnector agent variable to access its credentials secret
+	k8sAgentVar, err := variables.Get[apivariables.NutanixKonnectorAgent](
+		varMap,
+		v1alpha1.ClusterConfigVariableName,
+		"addons", v1alpha1.KonnectorAgentVariableName,
+	)
+	if err != nil {
+		return false, fmt.Errorf("failed to read konnector agent variable: %w", err)
+	}
+
+	if k8sAgentVar.Credentials == nil || k8sAgentVar.Credentials.SecretRef.Name == "" {
+		return false, fmt.Errorf("konnector agent credentials secret not configured")
+	}
+
+	// Get credentials from konnector agent addon Secret
+	credentialsSecret := &corev1.Secret{}
+	err = client.Get(ctx, types.NamespacedName{
+		Namespace: cluster.Namespace,
+		Name:      k8sAgentVar.Credentials.SecretRef.Name,
+	}, credentialsSecret)
+	if err != nil {
+		return false, fmt.Errorf("failed to get credentials secret: %w", err)
+	}
+
+	usernameData, ok := credentialsSecret.Data["username"]
+	if !ok {
+		return false, fmt.Errorf("credentials secret does not contain 'username' key")
+	}
+	passwordData, ok := credentialsSecret.Data["password"]
+	if !ok {
+		return false, fmt.Errorf("credentials secret does not contain 'password' key")
+	}
+
+	// Create credentials struct
+	credentials := prismgoclient.Credentials{
+		Endpoint: fmt.Sprintf("%s:%d", host, port),
+		URL:      fmt.Sprintf("https://%s:%d", host, port),
+		Username: string(usernameData),
+		Password: string(passwordData),
+		Insecure: prismCentralEndpointSpec.Insecure,
+		Port:     fmt.Sprintf("%d", port),
+	}
+
+	// Get kube-system namespace UUID from the cluster
+	clusterKey := ctrlclient.ObjectKeyFromObject(cluster)
+	remoteClient, err := remote.NewClusterClient(ctx, "", client, clusterKey)
+	if err != nil {
+		return false, fmt.Errorf("failed to create remote cluster client: %w", err)
+	}
+
+	kubeSystemNS := &corev1.Namespace{}
+	err = remoteClient.Get(ctx, types.NamespacedName{Name: "kube-system"}, kubeSystemNS)
+	if err != nil {
+		return false, fmt.Errorf("failed to get kube-system namespace from cluster(%s): %w", cluster.Name, err)
+	}
+
+	clusterUUID := string(kubeSystemNS.UID)
+
+	// Get trust bundle if insecure is false
+	var trustBundle string
+	if !prismCentralEndpointSpec.Insecure {
+		trustBundle = prismCentralEndpointSpec.AdditionalTrustBundle
+	}
```

Comment:
nit: it would be great to extract this is a separate util function (can be done in a follow up)